src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/PrisonerPatientUpdateService.kt

```diff
@@ -0,0 +1,119 @@
+package uk.gov.justice.digital.hmpps.prisontonhs.services
+
+import com.google.gson.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisontonhs.controllers.NhsPrisoner
+import uk.gov.justice.digital.hmpps.prisontonhs.repository.OffenderPatientRecord
+import uk.gov.justice.digital.hmpps.prisontonhs.repository.OffenderPatientRecordRepository
+import java.lang.reflect.Type
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import javax.persistence.EntityNotFoundException
+
+
+@Service
+class PrisonerPatientUpdateService(
+        private val offenderService: OffenderService,
+        private val prisonEstateService: PrisonEstateService,
+        private val nhsReceiveService: NhsReceiveService,
+        private val offenderPatientRecordRepository: OffenderPatientRecordRepository,
+        @Value("\${prisontonhs.only.prisons}") private val allowedPrisons: List<String>
+) {
+    companion object {
+        val log: Logger = LoggerFactory.getLogger(this::class.java)
+        val gson: Gson = GsonBuilder().setPrettyPrinting()
+                .registerTypeAdapter(LocalDate::class.java, LocalDateAdapter())
+                .create();
+    }
+
+    fun externalMovement(externalMovement: ExternalPrisonerMovementMessage) {
+
+        if (externalMovement.movementType in listOf("ADM", "REL")) {
```

Comment:
I suspect others may know what **ADM** and **REL** mean.  Sorry I don't.  I would find it helpful is these were pulled out into constants...

Reply:
done